model/riscv_sys_regs.sail

```diff
@@ -567,6 +567,26 @@ function retire_instruction() -> unit = {
        then minstret = minstret + 1
 }
 
+bitfield MEnvCfg : bits(64) = {
+  HADE : 61
+  /* Fill in other fields when adding support for them */
+}
+register menvcfg : MEnvCfg
+
+function legalize_menvcfg(c : MEnvCfg, high : bool, value : xlenbits) -> MEnvCfg = {
+  if high == false & sizeof(xlen) == 32 then
+    c
+  else {
+    // If 32bit, then high is true
+    let full : bits(64) = match (sizeof(xlen)) {
+      64 => value,
+      32 => value @ 0x0000_0000
+    };
+    let c = update_HADE(c, [full[61]]);
+    c
```

Comment:
I feel like there must be a better way to write all this... is this the first *h CSR we've implemented that's not just a stub 0?

Reply:
I can't say I'm anywhere close to a SAIL expert. I'm all ears on suggestions here. Thank you!